ray_lightning/ray_ddp.py

```diff
@@ -173,22 +172,7 @@ def setup(self):
             ray.get([w.execute.remote(self.init_hook) for w in self.workers])
 
     def setup_environment(self) -> None:
-        # Swap out the accelerator if necessary.
-        # This is needed to support CPU head with GPU workers or Ray Client.
-        current_accelerator = self.lightning_module.trainer.accelerator
-        if self.use_gpu and isinstance(current_accelerator, CPUAccelerator):
```

Comment:
I'm not quite following this logic - by removing `isinstance(current_accelerator, CPUAccelerator)`, what scenario does this solve? Wouldn't the problem case (`CPUAccelerator`) be changed to `DelayedGPUAccelerator` both before and after this PR?